src/ui/components/InterpretationState/InterpretationState.tsx

```diff
@@ -0,0 +1,319 @@
+        >
+          Verification Failed
+        </Button>
+      ) : mode === 'loading' ? (
+        <VStack gap={0}>
+          <Button
+            kind="regular"
+            size={36}
+            type="button"
+            style={{ position: 'relative' }}
+            onClick={() => loadingDialogRef.current?.showModal()}
+          >
+            <HStack gap={8} alignItems="center" justifyContent="center">
+              <CircleSpinner />
+              <LoadingText>Simulating</LoadingText>
+            </HStack>
+            <ZStack
+              hideLowerElements={true}
+              style={{
+                position: 'absolute',
+                top: 'calc(100% + 8px)',
+                whiteSpace: 'nowrap',
+                textAlign: 'left',
+              }}
+            >
+              <DelayedRender delay={11000}>
+                <span style={{ color: 'var(--black)' }}>
+                  (Going to give up soon...)
+                </span>
+              </DelayedRender>
+              <DelayedRender delay={6000}>
+                <span style={{ color: 'var(--black)' }}>
+                  (Request is taking longer than usual...)
+                </span>
+              </DelayedRender>
+            </ZStack>
+          </Button>
+          <DelayedRender delay={6000}>
+            <Spacer height={20} />
+          </DelayedRender>
+        </VStack>
+      ) : mode === 'error' ? (
+        <Button
+          kind="warning"
+          size={36}
+          type="button"
+          onClick={() => errorDialogRef.current?.showModal()}
+        >
+          <HStack gap={4} alignItems="center" justifyContent="center">
+            <ValidationErrorIcon
+              style={{ color: 'var(--notice-500)', width: 20, height: 20 }}
+            />
+            <span>Unverified</span>
+          </HStack>
+        </Button>
+      ) : mode === 'success' ? (
+        <Button
+          kind="regular"
+          size={36}
+          type="button"
+          onClick={() => {
+            warningSeverity === 'Green'
+              ? warningDialogRef.current?.showModal()
+              : successDialogRef.current?.showModal();
+          }}
+        >
+          <HStack gap={4} alignItems="center" justifyContent="center">
+            <ShieldIcon
+              style={{ color: 'var(--positive-500)', width: 20, height: 20 }}
+            />
+            <span>Verified</span>
+          </HStack>
+        </Button>
+      ) : null}
+      <BottomSheetDialog ref={loadingDialogRef} height="fit-content">
+        <InterpretationDescritionDialog mode="loading" />
+      </BottomSheetDialog>
+      <BottomSheetDialog ref={errorDialogRef} height="fit-content">
+        <InterpretationDescritionDialog mode="error" />
+      </BottomSheetDialog>
+      <BottomSheetDialog ref={successDialogRef} height="fit-content">
+        <InterpretationDescritionDialog mode="success" />
+      </BottomSheetDialog>
+      <BottomSheetDialog
```

Comment:
Why four differenet dialogs?

Reply:
This is intentional. To avoid content switching inside the dialog. If you click on Loading state, it will keep info about loading state.